magicctapipe/scripts/lst1_magic/semi_automatic_scripts/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,740 @@
+                        f'/fefs/onsite/common/MAGIC/data/M1/event/Calibrated/{i[0].split("_")[0]}/{i[0].split("_")[1]}/{i[0].split("_")[2]}/*{i[1]}.*_Y_*.root'
+                    )
+                    number_of_nodes = len(number_of_nodes) - 1
+
+                    with open(f"{source}_MAGIC-I_dl0_to_dl1_run_{i[1]}.sh", "w") as f:
+                        lines = [
+                            "#!/bin/sh\n\n",
+                            "#SBATCH -p long\n",
+                            f"#SBATCH -J {process_name}\n",
+                            f"#SBATCH --array=0-{number_of_nodes}\n",
+                            "#SBATCH -n 1\n\n",
+                            f"#SBATCH --output={target_dir}/{source}/DL1/Observations/M1/{i[0]}/{i[1]}/slurm-%x.%A_%a.out"
+                            f"#SBATCH --error={target_dir}/{source}/DL1/Observations/M1/{i[0]}/{i[1]}/slurm-%x.%A_%a.err"
+                            "ulimit -l unlimited\n",
+                            "ulimit -s unlimited\n",
+                            "ulimit -a\n\n",
+                            f"export OUTPUTDIR={target_dir}/{source}/DL1/Observations/M1/{i[0]}/{i[1]}\n",
+                            f"cd {target_dir}/{source}/../\n",
+                            "SAMPLE_LIST=($(<$OUTPUTDIR/list_dl0.txt))\n",
+                            "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n\n",
+                            "export LOG=$OUTPUTDIR/real_0_1_task_${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}.log\n",
+                            f"conda run -n {env_name} magic_calib_to_dl1 --input-file $SAMPLE --output-dir $OUTPUTDIR --config-file {target_dir}/{source}/config_DL0_to_DL1.yaml >$LOG 2>&1\n",
+                            "",
+                        ]
+                        f.writelines(lines)
+
+
+def directories_generator(
+    target_dir, telescope_ids, MAGIC_runs, NSB_match, source_name
+):
+
+    """
+    Here we create all subdirectories for a given workspace and target name.
+
+    Parameters
+    ----------
+    target_dir : str
+        Directory to store the results
+    telescope_ids : list
+        List of the telescope IDs (set by the user)
+    MAGIC_runs : array
+        MAGIC dates and runs to be processed
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    source_name : str
+        Name of the target source
+    """
+
+    if NSB_match:
+        if not os.path.exists(f"{target_dir}/v{__version__}"):
+            os.mkdir(f"{target_dir}/v{__version__}")
+        if not os.path.exists(f"{target_dir}/v{__version__}/{source_name}"):
+            os.mkdir(f"{target_dir}/v{__version__}/{source_name}")
+        if not os.path.exists(f"{target_dir}/v{__version__}/{source_name}/DL1"):
+            os.mkdir(f"{target_dir}/v{__version__}/{source_name}/DL1")
+        dl1_dir = str(f"{target_dir}/v{__version__}/{source_name}/DL1")
+    else:
+        if not os.path.exists(target_dir):
```

Comment:
again you can use os.makedirs
the current code will fail if target_dir was already created, but it did not have DL1 inside it